test/src/kbasesearchengine/test/parse/ObjectParserTest.java

```diff
@@ -0,0 +1,256 @@
+	private static Path scratch;
+	private static ObjectParser impl = null;
+
+	@BeforeClass
+	public static void init() throws Exception {
+		// config loading
+		String configFilePath = System.getenv("KB_DEPLOYMENT_CONFIG");
+		if (configFilePath == null) {
+			configFilePath = System.getProperty("KB_DEPLOYMENT_CONFIG");
+		}
+		File deploy = new File(configFilePath);
+		Ini ini = new Ini(deploy);
+		config = ini.get("KBaseSearchEngine");
+		scratch = Paths.get(config.get("scratch"), "test_object_parser");
+
+		impl = new ObjectParser();
+	}
+
+	@AfterClass
+	public static void teardown() throws Exception {
+		FileUtils.deleteDirectory(scratch.toFile());
+	}
+
+	@Test
+	public void prepareTempFileTest() throws Exception {
+
+		File tempDir = new File(scratch.toString(), UUID.randomUUID().toString());
+		assertEquals(false, tempDir.exists());
+
+		@SuppressWarnings("static-access")
+		File tmpFile = impl.prepareTempFile(tempDir);
+		assertEquals(true, tempDir.exists());
+		assertEquals(true, tmpFile.exists());
+		
+		assertThat(tmpFile.toString(), both(containsString("ws_srv_response_")).and(containsString(".json")));
+	}
+
+	@SuppressWarnings("static-access")
+	@Test
+	public void extractParentFragmentGenomeFeatureTest() throws Exception {
+
+		String jsonResource = "genome01";
+		String type = "GenomeFeature";
+
+		String parentJson = this.extractParentFragment(type, jsonResource);
+		String expectedParentJson = "{\"domain\":\"B\",\"scientific_name\":\"Shewanella\",\"assembly_ref\":\"1/2/1\"}";
+		assertEquals(parentJson, expectedParentJson);
+	}
+
+	@SuppressWarnings("static-access")
+	@Test
+	public void extractParentFragmentGenomeTest() throws Exception {
+
+		String jsonResource = "genome01";
+		String type = "Genome";
+
+		String parentJson = this.extractParentFragment(type, jsonResource);
+		assertNull(parentJson); // Genome.json doesn't have "from-parent" path
+	}
+
+	@SuppressWarnings("unchecked")
+	public static String extractParentFragment(String type, String jsonResource) throws Exception {
+
+		Map<String, Object> parsingRulesObj = UObject.getMapper()
+				.readValue(new File("resources/types/" + type + ".json"), Map.class);
+		ObjectTypeParsingRules parsingRules = ObjectTypeParsingRules.fromObject(parsingRulesObj, "test");
+		
+		try (JsonParser jts = SubObjectExtractorTest.getParsedJsonResource(jsonResource)) {
+			String parentJson = ObjectParser.extractParentFragment(parsingRules, jts);
+			return parentJson;
+		}
+	}
+
+	@SuppressWarnings("static-access")
+	@Test
+	public void prepareGUIDGenomeFeatureTest() throws Exception {
+
+		String jsonResource = "genome01";
+		String type = "GenomeFeature";
+
+		List<String> expectedIds = Arrays.asList("NewGenome.CDS.6211", "NewGenome.CDS.6210", "NewGenome.CDS.6209");
+
+		ArrayList<GUID> idList = this.prepareGUID(type, jsonResource, new GUID("WS:1/1/1"));
+
+		assertThat(idList.size(), is(3));
+		for (GUID guid : idList) {
+			assertThat(expectedIds, hasItem(guid.getSubObjectId()));
```

Comment:
It there a reason you're not checking the ordering of the results?

Reply:
👍